File: server/models/uml_diagram_class.py
from pydantic import BaseModel, Field
from typing import List


class Attribute(BaseModel):
    name: str
    type: str


class Method(BaseModel):
    name: str
    return_type: str


class Relationship(BaseModel):
    type: str
    target: str


class UMLDiagramClass(BaseModel):
    name: str
    attributes: list[Attribute]
    methods: list[Method]
    relationships: list[Relationship]


class UMLDiagramClasses(BaseModel):
    uml_diagram_class: List[UMLDiagramClass] = Field(default_factory=list)

    class Config:
        extra = "forbid"

File: client2/src/model/uml-diagram-class.ts
export type Attribute = {
    name: string;
    type: string;
};

export type Method = {
    name: string;
    return_type: string;
};

export type Relationship = {
    type: string;
    target: string;
};

export type UMLDiagramClass = {
    name: string;
    attributes: Attribute[];
    methods: Method[];
    relationships: Relationship[];
};

export type UMLDiagramClasses = {
    uml_diagram_class: UMLDiagramClass[];
};
